### ISSUE TITLE
Does this kind of usage of styled-components raise any concern?

### ISSUE THREAD

**Issue by marcusletric** [styled-components/styled-components#1765]
I have an interesting way of using styled-components, which seems to diverge from the guidelines.
Here is an example: 
``` 
const extraTaggedTemplate = ([...strings], ...expressions) => `${strings.map((string, index) => string + expressions[index])}`;

const fontSize = role => role === 'small' ? 18 : 20;

styled.a`${({role, color, theme}) =>`
       color: ${theme.palette[color]};
       ${extraTaggedTemplate`
             font-size: ${fontSize(role)}
       `}
`}`
```
This is just a rough example, NOT PRODUCTION CODE but what I want to highlight is that I have one huge function interpolated in the css string template as opposed to multiple smaller ones (as in your examples f.e. [Theming](https://www.styled-components.com/docs/advanced#theming)). Also I have multiple level of nested interpolations and tagged templates sometimes.

Basically I write the styles as they would be usual ES6 template strings.

Does this kind of usage raise any concern? I mean so far everything seems to work fine, even without using the [css helper](https://www.styled-components.com/docs/api#css). (The documentation says I need to use this when nesting interpolations, but so far I'm fine without it)

Is this a bad practice of using styled components? Should I be worried about any template string parsing behaviour on styled-components side that might surprise me on the longer run if I continue this approach?

**Comment by probablyup** [comment]
It's not a bad practice, but it will generate a ton of dynamic classNames since basically everything is a dynamic fragment. Especially if each interpolation has many possible values.

In general we recommend structuring your CSS to be as static as possible (read: not a lot of function interpolations), and layering in function interpolations sparingly to grab values from theme, etc. If you have any cases where the dynamic value changes a lot, like for example a color picker etc. we recommend that particular case be done as an inline style to reduce the amount of classes being generated.

Hope that helps!
